ports/raspberrypi/common-hal/rp2clock/OutputPin.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2024 Elliot Buller
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include "shared-bindings/microcontroller/Pin.h"
+#include "common-hal/rp2clock/OutputPin.h"
+#include "bindings/rp2clock/AuxSrc.h"
+#include "hardware/clocks.h"
+#include "hardware/gpio.h"
+#include "py/runtime.h"
+
+// GPOUT must be [21,23,24,25]
+void common_hal_rp2clock_outputpin_validate_src_pin(const mcu_pin_obj_t *pin) {
+    if ((pin->number != 21) && (pin->number != 23) &&
+        (pin->number != 24) && (pin->number != 25)) {
+        mp_raise_ValueError_varg(MP_ERROR_TEXT("Pin %d invalid, valid pins are: 21,23,24,25"), pin->number);
```

Comment:
I'd suggest just using https://github.com/adafruit/circuitpython/blob/8027efe24575fd4cfce0409c890348580ea291d2/shared-bindings/microcontroller/Pin.h#L45 because the pin names will vary per-board and the pin number may not be obvious.